Added Custom SVG Edge Path Section in Learn/Custom Edges Page (#662)
xyflow/web: sites/reactflow.dev/src/pages/learn/customization/custom-edges.mdx

custom node.

<RemoteCodeViewer route="learn/custom-edge-2" framework="react" />

## Making Custom SVG Edge Path

As discussed previously, if you want to make a custom edge in React Flow, you have to use either of the four path creation functions discussed above (e.g [`getBezierPath`](/api-reference/utils/get-bezier-path)). However if you want to make some other path shape like a Sinusoidal edge or some other edge type then you will have to make the edge path yourself. 

> The edge path we get from functions like [`getBezierPath`](/api-reference/utils/get-bezier-path) is just a path string which we pass into the `path` prop of the `<BaseEdge />` component. It contains the necessary information needed in order to draw that path, like where it should start from, where should it curve, where should it end etc. A simple straight path string between two points `(x1, y1)` to `(x2, y2)` would look like:

Reviewer: ```suggestion
The edge path we get from functions like [`getBezierPath`](/api-reference/utils/get-bezier-path) is just a path string which we pass into the `path` prop of the `<BaseEdge />` component. It contains the necessary information needed in order to draw that path, like where it should start from, where it should curve, where it should end, etc. A simple straight path string between two points `(x1, y1)` to `(x2, y2)` would look like:
```

Author: Hi @printerscanner Thanks for going through this PR and giving these suggestions. I will make the necessary changes.

```jsx
M x1 y1 L x2 y2
```

> A SVG path is basically a concatenated list of various commands like `M`, `L`, `Q` etc along with their values. Some of these commands are listed below along with their supported values

Reviewer: ```suggestion
An SVG path is a concatenated list of commands like M, L, Q, etc, along with their values. Some of these commands are listed below, along with their supported values.
```

- `M x1 y1` is the Move To command which moves the current point to the x1, y1 coordinate.
- `L x1 y1` is the Line To command which draws a line from the current point to x1, y1 coordinate.
- `Q x1 y1 x2 y2` is the Quadratic Bezier Curve command which draws a bezier curve from the current point to the x2, y2 coordinate. x1, y1 is the control point of the curve which determines the curviness of the curve. 

> Whenever we would want to start a path for our custom edge, we would use the `M` command to move our current point to `sourceX, sourceY` which we get as props in the custom edge component. Then based on the shape we want, we will use other commands like `L`(to make lines), `Q`(to make curves) and then finally end our path at `targetX, targetY` which we get as props in the custom edge component.

Reviewer: ```suggestion
Whenever we want to start a path for our custom edge, we use the `M` command to move our current point to `sourceX, sourceY` which we get as props in the custom edge component. Then based on the shape we want, we will use other commands like `L`(to make lines), `Q`(to make curves) and then finally end our path at `targetX, targetY` which we get as props in the custom edge component.
```

If you want to learn more about SVG paths, you can check out [SVG-Path-Editor](https://yqnn.github.io/svg-path-editor/). You can paste any SVG path there and analyze individual path commands via an intuitive UI.

> Here is an example with two types of custom edge paths, a Step edge and a Sinusoidal edge. You would be encouraged to look at the Step edge first to get your hands dirty with custom SVG paths since its simple and then look at how the Sinusoidal edge is made. After going through this example, you will have the necessary knowledge of making your own custom SVG paths for your custom edges.

Reviewer: ```suggestion
Here is an example with two types of custom edge paths, a Step edge and a Sinusoidal edge. You should look at the Step edge first to get your hands dirty with custom SVG paths since it's simple, and then look at how the Sinusoidal edge is made. After going through this example, you will have the necessary knowledge to make custom SVG paths for your custom edges.
```
